src/cmd/ksh93/tests/types.sh

```diff
@@ -29,7 +29,8 @@ Command=${0##*/}
 integer Errors=0
 
 tmp=$(mktemp -dt tmp.XXXXXXXXXX) || { err_exit mktemp -dt failed; exit 1; }
-trap "cd /; rm -rf $tmp" EXIT
+trap 'rm -fr $tmp' EXIT
+trap 'rm -fr $tmp; exit 1' 0 1 2 3 5 7 8 9 10 15
```

Comment:
This is completely unnecessary. If you do some experiments you should see that the parent shell is told the child shell died due to a signal. Most shells convert that into a status of 128 plus the signal number. We don't need to remove the temp file in this situation since it should never happen. And if it does happen we don't care that the temp file didn't get removed. This just adds unneeded and unwanted complexity.

Also, your commit comment implies this traps SIGABRT but on every system I've ever used that is signal 6 which is not in the list you're trapping.

P.S., You can't trap signal 9 (SIGKILL) and there is no signal zero. Ksh treats `0` as a synonym for `EXIT`. So this simply makes the preceding trap statement irrelevant.